### PR TITLE
`OfflineRecognizerResult` missing tokens, timestamps, lang, emotion, event

### PR DESCRIPTION
**Pull request by eschmidbauer**

### CODE REVIEW

File: scripts/go/sherpa_onnx.go [hunk 1]
```diff
@@ -593,7 +598,20 @@ func (s *OfflineStream) GetResult() *OfflineRecognizerResult {
 	defer C.SherpaOnnxDestroyOfflineRecognizerResult(p)
 	result := &OfflineRecognizerResult{}
 	result.Text = C.GoString(p.text)
-
+	result.Lang = C.GoString(p.lang)
+	result.Emotion = C.GoString(p.emotion)
+	result.Event = C.GoString(p.event)
+	n := int(p.count)
+	result.Tokens = make([]string, n)
+	tokens := (*[1 << 28]*C.char)(unsafe.Pointer(p.tokens_arr))[:n:n]
```
Comment (csukuangfj) [inline scripts/go/sherpa_onnx.go:606]: By the way, have you tested the changes?

Could you update the  examples to output the newly added fields?

Also, please consider the case where p.timestamps may be a nullptr.

**Review by csukuangfj — CHANGES_REQUESTED** [review]

File: scripts/go/sherpa_onnx.go (hunk 1 above)
Comment (csukuangfj) [reply scripts/go/sherpa_onnx.go:606]: Should it be

**C.char

Comment (eschmidbauer) [reply scripts/go/sherpa_onnx.go:606]: `(*[1 << 28]*C.char)` is correct based on what i have observed in code elsewhere

Comment (eschmidbauer) [reply scripts/go/sherpa_onnx.go:606]: > By the way, have you tested the changes?
> 
> Could you update the examples to output the newly added fields?
> 
> Also, please consider the case where p.timestamps may be a nullptr.

added check for null timestamps

**Comment by csukuangfj** [comment]
The CI failed. Could you have a look?

File: scripts/go/sherpa_onnx.go (hunk 1 above)
Comment (csukuangfj) [reply scripts/go/sherpa_onnx.go:606]: > `(*[1 << 28]*C.char)` is correct based on what i have observed in code elsewhere 
> 

Please test both methods and see which one is correct.

**Comment by csukuangfj** [comment]
By the way, you can test it locally.

You can replace the `sherpa_onnx.go` in your `go env GOPATH` sub-directory.

**Comment by eschmidbauer** [comment]
> By the way, you can test it locally.
> 
> You can replace the `sherpa_onnx.go` in your `go env GOPATH` sub-directory.

Thanks! Ill test it out today and post results/updates if needed

**Comment by eschmidbauer** [comment]
tested and confirmed this code works
`go.mod`
image


image


image

**Comment by csukuangfj** [comment]
Thank you for your contribution!

**Pull request merged**